### ISSUE TITLE
Clang ARM64 builds are defining -mthreads in wx-config, but that's an unsupported switch

### ISSUE THREAD

**Issue by hmartinez82** [wxWidgets/wxWidgets#23314]
### Description
In MSYS2's CLANGARM64 environment downstream deps are failing to build when they are restrictive towards unsupported command line switches. See https://github.com/msys2/MINGW-packages/issues/15825

I think that the issue comes from this line:
https://github.com/wxWidgets/wxWidgets/blob/master/configure.in#L4631
It's only honoring `-mthreads` not being supported in MINGW64, but it should also not add `-mthreads` when the host is `aarch64-w64-mingw32`

### Platform and version information

- wxWidgets version 3.2.2.1:
- wxMSW
- OS MSYS2 under Windows 11 10.0.22621
- Compiler: 
```
clang version 15.0.7
Target: aarch64-w64-windows-gnu
Thread model: posix
```

**Comment by hmartinez82** [comment]
Correct. When I meant `restrictive towards unsupported command line switches` that's exactly what I meant :)
KiCAD was such an example. I think that `-mthreads`` is only used in 32 bit x86? (Not exactly my expertise these command line switches)

**Comment by vadz** [comment]
It looks like this should have been closed by the commits above, but was just forgotten. Closing it now, please let me know if the problem still exists.
